File: Ser2023/Store/Store.Web/Modules/Store/ProductDetails/ProductDetailsForm.cs

namespace Store.Store.Forms
{
    using Serenity.ComponentModel;
    using Serenity.Web;

    [FormScript("Store.ProductDetails")]
    [BasedOnRow(typeof(Entities.ProductDetailsRow), CheckNames = true)]
    public class ProductDetailsForm
    {
        [EditLink, Width(200)]
        public int WaresId { get; set; }
        [Width(100)]
        [AlignRight, DisplayFormat("#,##0.0000")]
        public double Quantity { get; set; }
        [Width(100)]
        [AlignRight, DisplayFormat("#,##0.0000")]
        public double ProductQuantity { get; set; }
        [Width(100)]
        [AlignRight, DisplayFormat("#,##0.0000")]
        public decimal PlanPrice { get; set; }
        [Width(100)]
        [AlignRight, DisplayFormat("#,##0.0000")]
        public double Discount { get; set; }
    }
}

File: Ser2023/Company/Company.Web/Modules/ServerTypes/Store/ProductDetailsForm.ts
import { LookupEditor, StringEditor, DecimalEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface ProductDetailsForm {
    WaresId: LookupEditor;
    WaresName: StringEditor;
    Quantity: DecimalEditor;
    ProductQuantity: DecimalEditor;
    PlanPrice: DecimalEditor;
    Discount: DecimalEditor;
}

export class ProductDetailsForm extends PrefixedContext {
    static formKey = 'Store.ProductDetails';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ProductDetailsForm.init)  {
            ProductDetailsForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;
            var w2 = DecimalEditor;

            initFormType(ProductDetailsForm, [
                'WaresId', w0,
                'WaresName', w1,
                'Quantity', w2,
                'ProductQuantity', w2,
                'PlanPrice', w2,
                'Discount', w2
            ]);
        }
    }
}